dask_cuda/device_host_file.py

```diff
@@ -1,28 +1,39 @@
 from zict import Buffer, File, Func
 from zict.common import ZictBase
-from distributed.protocol import deserialize_bytes, serialize_bytes
+from distributed.protocol import (
+    deserialize,
+    deserialize_bytes,
+    serialize,
+    serialize_bytes,
+)
 from distributed.worker import weight
 
 from functools import partial
 import os
 
 from .is_device_object import is_device_object
+from .utils import move_frames_to_device, move_frames_to_host
 
 
-def _serialize_if_device(obj):
-    """ Serialize an object if it's a device object """
-    if is_device_object(obj):
-        return serialize_bytes(obj, on_error="raise")
-    else:
-        return obj
+def _serialize(obj):
+    """ Serialize an object and moves frames to host if it's a device object """
+    header, frames = serialize(
+        obj, serializers=["cuda", "dask", "pickle"], on_error="raise"
```

Comment:
Are there situations where we would use the dask or pickle serializers here?

Reply:
Yes, we also use this for serializing host objects into the `host_buffer`. But maybe "pickle" could be removed, is there any host-side objects that can't be serialized with "dask" serializer?

Reply:
Ah, I figured that this was just for `device<->host`.  Never mind then

Reply:
"dask" and "pickle" go together.  The "dask" serializer just special-cases a few host-side objects to get better performance.  There are many host-side objects that the "dask" serializer doesn't handle.

Reply:
Thanks for the clarification!